tests/unit/test_session.py

```diff
@@ -1624,6 +1624,150 @@ def test_stop_tuning_job_client_error(sagemaker_session):
     )
 
 
+def test_train_with_sagemaker_config_injection_empty_profiler_config(sagemaker_session):
+    """
+    Tests disableProfiler property is injected into a profiler_config object when the property does not exist explicitly
+    :param sagemaker_session:
+    :return: None
+    """
+    sagemaker_session.sagemaker_config = SAGEMAKER_CONFIG_TRAINING_JOB
+
+    in_config = [
+        {
+            "ChannelName": "training",
+            "DataSource": {
+                "S3DataSource": {
+                    "S3DataDistributionType": "FullyReplicated",
+                    "S3DataType": "S3Prefix",
+                    "S3Uri": S3_INPUT_URI,
+                }
+            },
+        }
+    ]
+
+    out_config = {"S3OutputPath": S3_OUTPUT}
+
+    resource_config = {
+        "InstanceCount": INSTANCE_COUNT,
+        "InstanceType": INSTANCE_TYPE,
+        "VolumeSizeInGB": MAX_SIZE,
+    }
+
+    stop_cond = {"MaxRuntimeInSeconds": MAX_TIME}
+    RETRY_STRATEGY = {"MaximumRetryAttempts": 2}
+    hyperparameters = {"foo": "bar"}
+    TRAINING_IMAGE_CONFIG = {
+        "TrainingRepositoryAccessMode": "Vpc",
+        "TrainingRepositoryAuthConfig": {
+            "TrainingRepositoryCredentialsProviderArn": "arn:aws:lambda:us-west-2:1234567897:function:test"
+        },
+    }
+
+    profiler_config = {}
```

Comment:
What is the default for `profiler_config` when nothing is provided? We know its None when we pass params to train() function but later it does do any defaults by setting some dict value? Worth checking that as well.

Reply:
Makes sense, I checked session.py and did not see any function implementation of the profiler_config object with function defaults - great callout and learnings from the DebugHookConfig. 

Let me know if there are any other places to check though

Reply:
since disable_profiler is set the False by default, the SDK should have some default profiler config it uses. But that might be before the session.py is called.

Mentioning and reopening this comment in case its relevant. Can close it again if its not relevant

Reply:
I double checked the session.py and confirmed that there is no custom dict being built, for estimator.py we inject only the disable_profiler value. I think we should be good - Thanks for the callout :)